packages/caravan-clients/src/client.test.ts

```diff
@@ -1823,4 +1825,327 @@ describe("BlockchainClient", () => {
       });
     });
   });
+  
+  
+  describe('getAddressTransactionHistory', () => {
+  const testAddress = 'bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh';
+  const testAddress2 = 'bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq';
+
+  beforeEach(() => {
+  vi.spyOn(wallet, 'callBitcoindWallet');
+  });
+
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  describe('parameter validation', () => {
+    it('should validate count parameter', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.MEMPOOL,
+        network: Network.MAINNET,
+      });
```

Comment:
you can move this into a beforeEach (or just create once in the `describe` block if these are all the same args)